src/solvers/strings/string_constraint_generator_main.cpp

```diff
@@ -114,6 +114,19 @@ void merge(string_constraintst &result, string_constraintst other)
     std::back_inserter(result.not_contains));
 }
 
+and_exprt char_set_restrictions(const exprt &expr, const std::string &char_set)
```

Comment:
This `char_set` format is a bit weird. It's not a regexp, and its not a string either.
That's not really your fault, but if we want to make `char_set_restrictions` work for other things than just to constraint to printable chars, it would be better for that function to take a more flexible format. I suggest giving a lower char and an upper char as parameter.